### PR TITLE
fix: senior deployer bug && added  FIAssessorFab

### PR DESCRIPTION
**Pull request by manuelpolzhofer**

### CODE REVIEW

File: src/core/lender/deployer.sol [hunk 1]
```diff
@@ -36,7 +37,7 @@ contract TrancheFab {
 
 contract SeniorTrancheFab {
     function newSeniorTranche(address currency, address token, address assessor) public returns (SeniorTranche tranche) {
-        tranche = new SeniorTranche(currency, token, assessor);
+        tranche = new SeniorTranche(token, currency, assessor);
```
Comment (manuelpolzhofer) [inline src/core/lender/deployer.sol]: I took me an hour to figure it out

Comment (charleenfei) [reply src/core/lender/deployer.sol]: thank you

File: src/core/test/system/base_system.sol [hunk 2]
```diff
@@ -57,24 +67,32 @@ contract BaseSystemTest is TestSetup, Math, DSTest {
...
         collateralNFT.transferFrom(sender, recipient, tokenId);
     }
 
     // helpers borrower
-    function issueNFT(address usr) public returns (uint, bytes32) {
-        uint tokenId = collateralNFT.issue(usr);
-        bytes32 lookupId = keccak256(abi.encodePacked(collateralNFT_, tokenId));
         // user approves shelf too lock NFT
+    function createSeniorInvestor() public {
+        seniorInvestor = new Investor(address(seniorOperator), currency_, address(seniorERC20));
```
Comment (charleenfei) [inline src/core/test/system/base_system.sol]: I will take out this code in my PR, I updated the createTestUsers method to handle this, but I think it's fine to leave in now

File: src/core/test/system/users/investor.sol [hunk 3]
```diff
@@ -30,6 +30,7 @@ contract Investor is DSTest {
 
     function doSupply(uint amount) public {
         address tranche_ = address(operator.tranche());
+        assertEq(currency.balanceOf(address(this)), amount);
```
Comment (charleenfei) [inline src/core/test/system/users/investor.sol]: why do they need to supply the entirely of their balance every time?

Comment (manuelpolzhofer) [reply src/core/test/system/users/investor.sol]: Yes, I wanted to add greater than

**Pull request merged**